### PR TITLE
Add initial content outline

### LINKED ISSUE #13: Publish initial content outline.
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Description of request

Publicly develop and publish an initial content outline around which to organize content creation. Leverage this structure to begin creating issues focusing on specific content areas.

### Related link(s)

_No response_

### Are you interested in authoring this content?

- [ ] Yes

### PR DESCRIPTION
**Pull request by justinfagnani**
Fixes #13 

This is preliminary - I added a bunch of sections I thought reasonable, but it's all up for changes.

I think we should emphasize that the outline is a living document and probably spend some amount of time massaging it here, but not try to get it perfect right away and iterate in PRs.

We should probably also add a note about our intended workflow: like if you pick a page to write, make an issue and link to it from the outline? We could also proactively make a bunch of issues, but that might require more synchronization work as the outline changes. Once we get content in place, if we make a change to the outline we'll want to know where it's out of sync and needs content changes to bring it into sync - or, once we get enough content we retire this document in favor of the actual content structure.

### CODE REVIEW

**Comment by castastrophe** [comment]
👍 Looks great!

**Review by castastrophe — APPROVED** [review]

File: plan/outline.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  * Styling
+    * Encapsulation (more detail than)
+    * Shadow selectors: `:host`, `:host()`, and `::slotted()`
+    * CSS Custom Properties
+    * CSS Shadow Parts: `part` and `::part()`
+    * Constructible style sheets
+    * Theming
+      * Strategies for deep styling
+  * Dealing with data
+    * Making reactive properties with accessors
```
Comment (kevinpschaaf) [inline plan/outline.md:58]:
```suggestion
    * Designing both a property & attribute API
    * Responding to attribute changes (`attributeChangedCallback`)
    * Responding to property changes (accessors)
    * Considerations about when to use methods
```

File: plan/outline.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    * Reflecting properties to attributes
```
Comment (kevinpschaaf) [inline plan/outline.md:59]: ...and vice-versa
```suggestion
    * Reflecting between properties and attributes
```

File: plan/outline.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    * Slots
+    * Communicating with events
+    * Using children
+  * Accessibility
+  * Forms
+  * Web component libraries
+* Community Protocols
+* Community and Resources
+* FAQ
+  * Myth-busting answers can go here ("can custom elements really only take strings?", etc)
```
Comment (kevinpschaaf) [inline plan/outline.md:72]: Brainstormed a few more ideas here...
```suggestion
  * Myth-busting answers can go here
    * Can custom elements really only take strings?
    * Is it impossible to SSR web components?
    * Why don't web components solve ___ that other frameworks do?
    * ...
```

**Review by kevinpschaaf — APPROVED** [review]

**Comment by kevinpschaaf** [comment]
Looks good, just a few suggestions.

File: plan/outline.md (hunk 1 above)
Comment (justinfagnani) [reply plan/outline.md:58]: I meant to call out accessors specifically here as that's a key technique to have reactive state outside of attributes.

File: plan/outline.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    * CSS Custom Properties
+    * ::part()s
+  * Using web components with frameworks
+  * Using web components in...
+    * Markdown (how to import in various systems)
+    * WordPress
+* Writing web components
+  * Defining
+    * Custom element registries
+    * Self-registering vs not
```
[the remark this answers is not in the document]
Comment (justinfagnani) [reply plan/outline.md:38]: Right, that's what I intend. There are patterns of publishing a self-registering module _and_ a non-registering module to be forward compatible with scoped registries, or allow extension w/o registering the base class. I think it's worth a mention - along with the risks of not self-registering.

File: plan/outline.md (hunk 1 above)
Comment (justinfagnani) [reply plan/outline.md:58]: oh - I misread the diff... you just expanded it :)

**Pull request merged**